src/test/java/seedu/address/storage/JsonAdaptedExpenseTest.java

```diff
@@ -29,28 +30,31 @@ public void toModelType_validExpenseDetails_returnsExpense() throws Exception {
     @Test
     public void toModelType_invalidAmount_throwsIllegalValueException() {
         JsonAdaptedExpense expense =
-                new JsonAdaptedExpense(VALID_DESCRIPTION, VALID_EXPENSE_CATEGORY, INVALID_AMOUNT);
+                new JsonAdaptedExpense(VALID_DESCRIPTION, VALID_EXPENSE_CATEGORY, INVALID_AMOUNT, VALID_EXPENSE_DATE);
         String expectedMessage = Amount.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, expense::toModelType);
     }
 
     @Test
     public void toModelType_nullDescription_throwsIllegalValueException() {
-        JsonAdaptedExpense expense = new JsonAdaptedExpense(null, VALID_EXPENSE_CATEGORY, VALID_AMOUNT);
+        JsonAdaptedExpense expense = new JsonAdaptedExpense(null, VALID_EXPENSE_CATEGORY,
+                VALID_AMOUNT, VALID_EXPENSE_DATE);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Description.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, expense::toModelType);
     }
 
     @Test
     public void toModelType_nullExpenseCategory_throwsIllegalValueException() {
-        JsonAdaptedExpense expense = new JsonAdaptedExpense(VALID_DESCRIPTION, null, VALID_AMOUNT);
+        JsonAdaptedExpense expense = new JsonAdaptedExpense(VALID_DESCRIPTION, null, VALID_AMOUNT,
+                VALID_EXPENSE_DATE);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, ExpenseCategory.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, expense::toModelType);
     }
 
     @Test
     public void toModelType_nullAmount_throwsIllegalValueException() {
-        JsonAdaptedExpense expense = new JsonAdaptedExpense(VALID_DESCRIPTION, VALID_EXPENSE_CATEGORY, null);
+        JsonAdaptedExpense expense = new JsonAdaptedExpense(VALID_DESCRIPTION, VALID_EXPENSE_CATEGORY, null,
+                VALID_EXPENSE_DATE);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Amount.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, expense::toModelType);
     }
```

Comment:
I think can add additional test cases for invalid date / null date! Can keep in mind for v1.3!